ext/native.go

```diff
@@ -704,21 +704,16 @@ func (t *nativeType) hasField(fieldName string) (reflect.StructField, bool) {
 }
 
 func adaptFieldValue(adapter types.Adapter, refField reflect.Value) ref.Val {
-	return adapter.NativeToValue(getFieldValue(adapter, refField))
+	return adapter.NativeToValue(getFieldValue(refField))
 }
 
-func getFieldValue(adapter types.Adapter, refField reflect.Value) any {
+func getFieldValue(refField reflect.Value) any {
 	if refField.IsZero() {
 		switch refField.Kind() {
-		case reflect.Array, reflect.Slice:
-			return types.NewDynamicList(adapter, []ref.Val{})
-		case reflect.Map:
-			return types.NewDynamicMap(adapter, map[ref.Val]ref.Val{})
 		case reflect.Struct:
 			if refField.Type() == timestampType {
 				return types.Timestamp{Time: time.Unix(0, 0)}
 			}
-			return reflect.New(refField.Type()).Elem().Interface()
```

Comment:
It's probably worth noting how this differs from the `nativeToValue` behavior in `native.go`. The `time.Unix(0, 0)` value can be returned on its own in the timestamp block since the proto semantics are that uninitialized timestamp accesses are treated as unix time 0. 

``` go
// Proto semantics will return a time value initialized at Unix time 0 if the value is unset,
// but golang will return the zero value for `time.Time` which is not the same.
if refField.Type() == timestampType {
	return time.Unix(0, 0)
}
```

Reply:
Updated to now simply return `time.Unix(0,0)` 👍